src/docbuilder/queue.rs

```diff
@@ -63,74 +66,22 @@ impl DocBuilder {
         Ok(crates_added)
     }
 
-    pub fn get_queue_count(&self) -> Result<i64> {
-        let conn = self.db.get()?;
-
-        Ok(conn
-            .query("SELECT COUNT(*) FROM queue WHERE attempt < 5", &[])?
-            .get(0)
-            .get(0))
-    }
-
     /// Builds the top package from the queue. Returns whether there was a package in the queue.
     ///
     /// Note that this will return `Ok(true)` even if the package failed to build.
     pub(crate) fn build_next_queue_package(
         &mut self,
         builder: &mut RustwideBuilder,
     ) -> Result<bool> {
-        // This is in a nested scope to drop the connection before build_package is called,
-        // otherwise the borrow checker will complain.
-        let (id, name, version): (i32, String, String) = {
-            let conn = self.db.get()?;
-
-            let query = conn.query(
-                "SELECT id, name, version
-                 FROM queue
-                 WHERE attempt < 5
-                 ORDER BY priority ASC, attempt ASC, id ASC
-                 LIMIT 1",
-                &[],
-            )?;
-
-            if query.is_empty() {
-                // nothing in the queue; bail
-                return Ok(false);
-            }
+        let mut processed = false;
+        let queue = self.build_queue.clone();
```

Comment:
Is this clone necessary?

Reply:
Yes, because the closure captures `self`.